### PR TITLE
fix: recover scheduled chat failures

### PR DESCRIPTION
**Pull request by JeffOtano**
## Summary

- Wrap scheduled chat actions (`processMessage`, `continueAfterApproval`) in try/catch so pre-stream failures (provider resolution, prompt build, sanitized BYOK errors) persist as assistant messages instead of dying silently on the scheduler.
- Extract helpers to `chatHelpers.ts` and gate Discord notifications so expected user-state errors (missing key, missing model, house-key cap, classified provider errors) do not page ops.

## Changes

- `convex/chat.ts`: wrap both actions in try/catch; on error route through `persistScheduledFailure`. User message is still pre-saved before the guarded region so retries keep `promptMessageId`.
- `convex/chatHelpers.ts`: new `getScheduledFailureContent`, `shouldNotifyScheduledFailure`, `persistScheduledFailure`. Sentinel codes (`house_key_quota_exhausted`, `byok_key_missing`, `byok_model_missing`) are checked before BYOK classification to avoid substring collisions. Discord notification suppressed for expected error codes and classified provider errors.
- `convex/chatHelpers.test.ts`: unit tests covering error-code → user message mapping (sentinels, sanitized BYOK codes with/without provider, raw provider errors, generic fallback) and notification gating.

## Checklist

- [x] `npx tsc --noEmit` passes
- [x] `npm run lint` passes
- [x] `npm test` passes (existing + new tests)
- [x] New logic has tests (happy path + at least one error/edge case)
- [x] No new or materially expanded file exceeds the 300-line soft cap
- [x] No file exceeds the 400-line hard cap (enforced by CI)
- [x] Functions stay near the 60-line convention
- [x] No new `any` types (enforced by ESLint); `as` casts only at deserialization boundaries
- [ ] Tested in browser (if UI changes) — N/A, backend only
- [x] Commits follow conventional format (`type: description`)
- [x] No unused exports or dead code introduced

## Test Plan

- Unit tests in `convex/chatHelpers.test.ts` cover each error-code branch and the notification gate.
- Manual verification relied on existing `npm test` and typecheck; not exercised against live scheduler in this PR.

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced error handling and recovery in chat operations to persist failures and provide meaningful error messages to users
  * Improved error messaging for provider configuration issues and quota exhaustion scenarios
  * Refined error notification system to suppress unnecessary alerts for expected failures

* **Tests**
  * Added comprehensive test suite for error message generation and notification logic

### CODE REVIEW

File: convex/chat.ts [hunk 1]
```diff
@@ -196,30 +198,36 @@ export const continueAfterApproval = action({
...
-    });
-    await analytics.flush();
+      analytics.capture(userId, "coach_response_received", {
+        response_time_ms: Date.now() - startTime,
+        after_approval: true,
+      });
+      await analytics.flush();
+    } catch (error) {
+      await persistScheduledFailure({ ctx, threadId, userId, error, provider });
+    }
```
[the remark this answers is not in the document]
Comment (JeffOtano) [reply convex/chat.ts]: Applied. Hoisted `startTime` out of the try in both `continueAfterApproval` and `processMessage`, moved analytics after the catch, and `return` from the catch so analytics never runs on failure.

File: convex/chatHelpers.ts [hunk 2]
```diff
@@ -74,6 +76,80 @@ export async function withByokErrorSanitization<T>(fn: () => Promise<T>): Promis
...
+
+  if (!shouldNotifyScheduledFailure(args.error)) return;
+
+  const reason = args.error instanceof Error ? args.error.message : String(args.error);
+  await args.ctx.runAction(internal.discord.notifyError, {
+    source: "chat.processMessage",
+    message: reason,
+    userId: args.userId,
+  });
+}
```
[the remark this answers is not in the document]
Comment (JeffOtano) [reply convex/chatHelpers.ts:152]: Applied. `persistScheduledFailure` now takes a required `source` arg; callers pass `chat.processMessage` and `chat.continueAfterApproval`.

**Comment by JeffOtano** [comment]
Addressed review in 1953164:

**Applied**
- Analytics outside the guarded try (hoisted `startTime`, `return` in catch) in both actions
- `persistScheduledFailure` now takes a required `source` arg; callers pass `chat.processMessage` / `chat.continueAfterApproval`
- Dropped redundant `?? undefined` on `imageStorageIds`
- Added test pinning generic-fallback behavior for classifiable errors without a provider

**Skipped with rationale**
- Wrapping the pre-save `saveMessage` in its own try/catch: `persistScheduledFailure` itself calls `saveMessage`, so a DB-down scenario that fails the pre-save would also fail the fallback path. Scheduler retry already handles transient DB blips.
- BYOK fallback in the no-provider branch: in practice `provider` is only `undefined` when `resolveUserProviderConfig` throws sentinel codes (`byok_key_missing`, etc.), never raw BYOK errors - the branch is unreachable. It would also invalidate the new test pinning current behavior.
- Non-blocking Discord notify: already non-blocking. `postToDiscord` (convex/discord.ts:28-38) wraps fetch in try/catch and swallows errors, so `notifyError` cannot throw.
